synapse/federation/transaction_queue.py

```diff
@@ -239,6 +239,21 @@ def handle_room_events(events):
                     "events", next_token
                 )
 
+                synapse.metrics.event_processing_positions.set(
```

Comment:
can we move the `events_processed_counter` update here so that all the metrics updates are collected together?